generator.js

```diff
@@ -0,0 +1,23 @@
+ChangeBodyColor();
+
+function ChangeBodyColor () {
+  var stop = false;
+  document.body.style.backgroundColor = 'rgb(' + (Math.floor(Math.random() * 256)) + ',' + (Math.floor(Math.random() * 256)) + ',' + (Math.floor(Math.random() * 256)) + ')';
+  var startTimer = setTimeout(function() {
+  stop = true;
+  clearInterval(startTimer);
+  }, 10000);
+
+  changeColor();
+
+  function changeColor () {
+    var clearId = setTimeout(function() {
+      if (stop) {
+        clearInterval(clearId);
+        return;
+      }
+     document.body.style.backgroundColor = 'rgb(' + (Math.floor(Math.random() * 256)) + ',' + (Math.floor(Math.random() * 256)) + ',' + (Math.floor(Math.random() * 256)) + ')';
```

Comment:
It looks like this changeColor function is handling several concerns: setting/clearing intervals, creating an RGB color, and setting the background color. How could we separate these concerns?